docs/docs/design/remote-funcitons.md

```diff
@@ -0,0 +1,85 @@
+# Remote Functions
+
+Remote functions are part of the DXOS extenisbility framework that enables developers to deploy long-running agents and invoke short-lived functions that have access to Spaces.
+
+## Spec
+
+- Run long-lived functions that operate over users' ECHO spaces and interace with external systems.
+- Isolation of individual functions.
+- Controlled access to resources (e.g., storage, network). Enables "pure" stateless funcitons.
+- Functions triggered by external events (e.g., signaling, HALO/ECHO mutations, timers, other functions).
+- Global registration and discovery (DMG).
```

Comment:
use a scenario to make it clear why this is a thing?
e.g.: users can delegate invocation to functions they don't control for security and compliance purposes (auth flows, signing flows, on-behalf-of authorizations)? (I don't think this is a scenario we're interested in here though)